pkg/provider/onepassword/onepassword.go

```diff
@@ -152,13 +173,164 @@ func validateStore(store esv1beta1.GenericStore) error {
 	return nil
 }
 
-func (provider *ProviderOnePassword) DeleteSecret(_ context.Context, _ esv1beta1.PushSecretRemoteRef) error {
-	return fmt.Errorf("not implemented")
+func deleteField(fields []*onepassword.ItemField, label string) (fieldsF []*onepassword.ItemField, err error) {
+	// This will always iterate over all items
+	// but its done to ensure that two fields with the same label
+	// exist resulting in undefined behavior
+	var found bool
+	for _, item := range fields {
+		if item.Label == label {
+			if found {
+				return nil, ErrExpectedOneField
+			}
+			found = true
+			continue
+		}
+		fieldsF = append(fieldsF, item)
+	}
+	if !found {
+		return fieldsF, ErrNoChanges
+	}
+	return fieldsF, nil
+}
```

Comment:
```suggestion
func deleteField(fields []*onepassword.ItemField, label string) ([]*onepassword.ItemField, error) {
	// This will always iterate over all items
	// but its done to ensure that two fields with the same label
	// exist resulting in undefined behavior
	var fieldsF []*onepassword.ItemField
	var found bool
	for _, item := range fields {
		if item.Label == label {
			if found {
				return nil, ErrExpectedOneField
			}
			found = true
			continue
		}
		fieldsF = append(fieldsF, item)
	}
	if !found {
		return fieldsF, ErrNoChanges
	}
	return fieldsF, nil
}
```

Reply:
Actually since `ErrNoChanges` is ignored, might as well return `nil` here and then in case of a nil fieldsF just return nil. That way, we avoid having to deal with sentinel errors.